crates/autopilot/src/limit_orders/quoter.rs

```diff
@@ -58,10 +58,13 @@ impl LimitOrderQuoter {
 
     /// Returns whether it is likely that there is no more work.
     async fn update(&self) -> Result<bool> {
-        let order_specs = self
+        let orders = self
             .database
-            .order_specs_with_outdated_fees(self.limit_order_age, self.parallelism)
+            .open_limit_orders(self.limit_order_age)
             .await?;
+        let orders = self.orders_with_sufficient_balance(orders).await;
+        let order_specs: Vec<_> = orders.into_iter().map(order_spec_from).unique().collect();
+
```

Comment:
This changes semantics in a pretty big way, we used to get order_parameters_with_most_outdated_fees which was limited to `self.parallelism` orders and ordered by how long the order has been waiting for a price.

Now, we are getting much more orders and trying to update them all in an arbitrary order.

I would propose to return `database::orders::open_limit_orders` ordered ascending by `surplus_fee_timestamp` and to `.take(self.parallelism)`. `Itertools::unique` is stable, so the `order_specs` would retain the sorting from `open_limit_orders`.